### ISSUE TITLE
`ValidatingAdmissionPolicyBindingList` is always returned with non-zero length

### ISSUE THREAD

**Issue by Danil-Grigorev** [kubernetes/kubernetes#126739]
### What happened?

Performing request on `ValidatingAdmissionPolicyList` or `ValidatingAdmissionPolicyBindingList` returns a non-zero length response, event if there are no `ValidatingAdmissionPolicy` in the cluster.

### What did you expect to happen?

Return empty list, as other resource types.

### How can we reproduce it (as minimally and precisely as possible)?

Kubernetes v1.30
```
kubectl get ValidatingAdmissionPolicyBinding -A -o yaml
apiVersion: v1
items:
- apiVersion: admissionregistration.k8s.io/v1
  kind: ValidatingAdmissionPolicyBindingList
  metadata:
    resourceVersion: "29454"
kind: List
metadata:
  resourceVersion: ""
```

### Kubernetes version

<details>

```console
$ kubectl version

warning: This version information is deprecated and will be replaced with the output from kubectl version --short.  Use --output=yaml|json to get the full version.
Client Version: version.Info{Major:"1", Minor:"27", GitVersion:"v1.27.0", GitCommit:"1b4df30b3cdfeaba6024e81e559a6cd09a089d65", GitTreeState:"clean", BuildDate:"2023-04-11T17:10:18Z", GoVersion:"go1.20.3", Compiler:"gc", Platform:"linux/amd64"}
Kustomize Version: v5.0.1
Server Version: version.Info{Major:"1", Minor:"30", GitVersion:"v1.30.0", GitCommit:"7c48c2bd72b9bf5c44d21d7338cc7bea77d0ad2a", GitTreeState:"clean", BuildDate:"2024-05-13T22:00:36Z", GoVersion:"go1.22.2", Compiler:"gc", Platform:"linux/amd64"}
WARNING: version difference between client (1.27) and server (1.30) exceeds the supported minor version skew of +/-1
```

</details>

### Cloud provider

<details>

</details>

### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
# paste output here
$ uname -a
# paste output here

# On Windows:
C:\> wmic os get Caption, Version, BuildNumber, OSArchitecture
# paste output here
```

</details>

### Install tools

<details>

</details>

### Container runtime (CRI) and version (if applicable)

<details>

</details>

### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>

**Comment by xyz-li** [comment]
This has been fixed in [v1.30.3](https://github.com/kubernetes/kubernetes/releases/tag/v1.30.3)
https://github.com/kubernetes/kubernetes/pull/126153

**Comment by Danil-Grigorev** [comment]
@xyz-li Thanks, I can confirm. Closing this

**Comment by xyz-li** [comment]
> This has been fixed in [v1.30.3](https://github.com/kubernetes/kubernetes/releases/tag/v1.30.3)

I have confirmed this has been fixed in v1.30.4, not in v1.30.3.
